### ISSUE TITLE
TMSH CLI script can fail under load

### ISSUE THREAD

**Issue by dstokesf5** [F5Networks/f5-appsvcs-extension#581]
### Environment
 * Application Services Version: 3.35.0
 * BIG-IP Version: 15.1.5

### Summary
When a BIG-IP is under load, such as when doing a Prometheus pull from the Telemetry Streaming extension, AS3 can encounter issues executing the TMSH CLI script to apply changes. Below are two log samples of problems with the script.

#### Timeout
[{\"code\":422,\"message\":\"declaration failed\",\"response\":\"connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1644237601034313 checking cli script status timed out\",\"host\":\"localhost\",\"tenant\":\"Common\",\"runTime\":257363},{\"code\":422,\"message\":\"declaration failed\",\"response\":\"connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1644237861517075 checking cli script status timed out\",\"host\":\"localhost\",\"tenant\":\"Common\",\"runTime\":260588}],

#### Lost task ID
Mon, 07 Mar 2022 21:23:33 GMT - severe: [appsvcs] {"message":"Declaration failed: GET http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1646688141254309 checking cli script status response=404 body={\"code\":404,\"message\":\"Task not found - [ID: 1646688141254309](https://bugzilla.olympus.f5net.com/show_bug.cgi?id=1646688141254309) user: admin\",\"errorStack\":[],\"apiError\":1}","level":"error"}

### Steps To Reproduce
1. Setup a simple TS Prometheus configuration
```json
{
    "class": "Telemetry",
    "My_Poller": {
        "class": "Telemetry_System_Poller",
        "interval": 0
    },
    "My_System": {
        "class": "Telemetry_System",
        "enable": "true",
        "systemPoller": "My_Poller"
    },
    "My_Pull_Consumer": {
        "class": "Telemetry_Pull_Consumer",
        "type": "Prometheus",
        "systemPoller": "My_Poller"
    }
}
```

2. Pull Prometheus every 30 seconds
```bash
for x in {0..200}; do curl -sku admin:admin https://192.0.2.0mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer > /dev/null ; echo "pull iteration $x"; sleep 30; done
```

3. Continue POSTing AS3 declarations until errors are seen in  /var/log/restnoded/restnoded.log. This may require having multiple WAF policies to increase the time needed to run the CLI script.

### Expected Behavior
AS3 should be able to handle timeouts and other problems caused by system load while monitoring the status of the TMSH CLI script. If an unrecoverable error occurs, then AS3 should better inform the user of the problem.

### Actual Behavior
Currently error messages are passed directly to the user instead of trying to work around them or provide sufficient information to describe why the task could not be completed.

**Comment by dstokesf5** [comment]
This is in our internal product backlog as AUTOTOOL-3058.

**Comment by dstokesf5** [comment]
We are implementing some retry logic when the request to fetch the TMSH CLI script task times out.

For the missing task, we have found that increasing time outs to around 600 prevents this issue from occurring: https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/userguide/best-practices.html#increase-timeout-values-if-the-rest-api-is-timing-out. There is nothing we can do in AS3 to further prevent or resolve this issue. However, we are updating the error message to be more helpful in this case.

**Comment by ghalevy** [comment]
@dstokesf5 we ran into a very similar issue today but not under load. 

```
    "results": [
        {
            "code": 200,
            "message": "success",
            "lineCount": 117,
            "host": "localhost",
            "tenant": "Common",
            "runTime": 266247
        },
        {
            "code": 422,
            "message": "declaration failed",
            "response": "GET http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1649456801783336 checking cli script status response=404 body={\"code\":404,\"message\":\"Task not found - ID: 1649456801783336 user: admin\",\"errorStack\":[],\"apiError\":1}",
            "host": "localhost",
            "tenant": "Common",
            "runTime": 240480
        }
    ],
```

**Comment by dstokesf5** [comment]
This was resolved in the 3.36.0 release of AS3.
